spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java

```diff
@@ -218,14 +224,44 @@ public ConsulDiscoveryProperties(InetUtils inetUtils) {
 	}
 
 	/**
-	 * @param serviceId The service who's filtering tag is being looked up
-	 * @return The tag the given service id should be filtered by, or null.
+	 * Gets the tag to use when looking up the instances for a particular service. If the
+	 * service has an entry in {@link #serverListQueryTags} that will be used. Otherwise
+	 * the content of {@link #defaultQueryTag} will be used.
+	 * @param serviceId the service whose instances are being looked up
+	 * @return the tag to filter the service instances by or null if no tags are
+	 * configured for the service and the default query tag is not configured
 	 */
 	public String getQueryTagForService(String serviceId) {
 		String tag = this.serverListQueryTags.get(serviceId);
 		return tag != null ? tag : this.defaultQueryTag;
 	}
 
+	/**
+	 * Gets the array of tags to use when looking up the instances for a particular
+	 * service. If the service has an entry in {@link #serverListQueryTags} that will be
+	 * used. Otherwise the content of {@link #defaultQueryTag} will be used. This differs
+	 * from {@link #getQueryTagForService(String)} in that it assumes the configured tag
+	 * property value may represent multiple tags when separated by commas. When the tag
+	 * property is set to a single tag then this method behaves identical to its
+	 * aforementioned counterpart except that it returns a single element array with the
+	 * single tag value.
+	 * <p>
+	 * The expected format of the tag property value is {@code tag1,tag2,..,tagN}.
+	 * Whitespace will be trimmed off each entry.
+	 * @param serviceId the service whose instances are being looked up
+	 * @return the array of tags to filter the service instances by - it will be null if
+	 * no tags are configured for the service and the default query tag is not configured
+	 * or if a single tag is configured and it is the empty string
+	 */
+	@Nullable
+	public String[] getQueryTagsForService(String serviceId) {
+		String queryTagStr = getQueryTagForService(serviceId);
+		if (queryTagStr == null || queryTagStr.isEmpty()) {
```

Comment:
This allows us to knockout a previously set default during tests or in a profile specific property override.  Imagine the case where the default profile yaml section sets 
```java
spring.cloud.consul.discovery.default-query-tag: tag1
```
but we want to disable the tag filtering in qa env (or in a test) with 
```java
spring.cloud.consul.discovery.default-query-tag:
```
Prior to this this change, that would not work.